### PR TITLE
RelMon histogram comparison: adapt ranges in a compatible way with the definition of getNbins

### PR DESCRIPTION
**Pull request by fabiocos**
#### PR description:

The RelMon DQM histogram comparison is complaining about differences between histograms when comparing a file with itself, which is clearly incorrect, see the discussion in https://github.com/cms-sw/cmssw/issues/26496 . This issue looks linked to the inconsistent use in the utility package https://cmssdt.cern.ch/lxr/source/Utilities/RelMon/python/utils.py of number of bins and ranges. In python the range function has as a second argument not the last value of the list but the next to last. This looks likely the motivation for the function getNbins adding 1 unit to the number of bins as returned by the ROOT function getNbinsX(). There seems to be also some inconsistency between the starting point of the different range loops, in several cases bin 0 is used (underflow), sometime bin 1 (the first histogram bin). 

This PR tries to solve the issue, having all the range based loops running from the underflow (bin=0) to the overflow (bin=GetNbinsX()+1), which means setting in the range loops as last item GetNbinsX()+2. 

#### PR validation:

Running the test BinToBin to compare a file with itself does not show any more the problem. Using the DQM output of the generator workflow 503 with a syntax similar to cms-bot:
```
13:51 cmsdev25 4149> compare_using_files.py -B DQM/TimerService@3 -o out -C -R -p --no_successes -s b2b 503.0_MinBias_13TeV_pythia8+MinBias_13TeV_pythia8+HARVESTGEN/DQM_V0001_R000000001__Global__CMSSW_X_Y_Z__RECO.root 503.0_MinBias_13TeV_pythia8+MinBias_13TeV_pythia8+HARVESTGEN/DQM_V0001_R000000001__Global__CMSSW_X_Y_Z__RECO.root
Analysing Histograms located in directory  at: 
 o 503.0_MinBias_13TeV_pythia8+MinBias_13TeV_pythia8+HARVESTGEN/DQM_V0001_R000000001__Global__CMSSW_X_Y_Z__RECO.root
 o 503.0_MinBias_13TeV_pythia8+MinBias_13TeV_pythia8+HARVESTGEN/DQM_V0001_R000000001__Global__CMSSW_X_Y_Z__RECO.root
We have a Blacklist:
 o Directory TimerService at level 3
++++++++++++++++++++++++++++++
Output Directory will be  out
Studying directory Info, 1/2
Studying directory Generator, 2/2
 ->Appending Generator... Appended.
Finished

 - summary of 840 tests:
 o Failiures: 0.00% (0/840)
 o Nulls: 0.00% (0/840) 
 o Successes: 100.00% (840/840) 
 o Skipped: 0.00% (0/840) 
 o Missing objects: 0
Pickleing the directory as out.pkl in dir /build/fabiocos/106X/relmon/CMSSW_10_6_X_2019-04-29-2300/work/out
Reading directory from out.pkl
Calculating stats for the directory...
Producing html...
```

### CODE REVIEW

**Comment by fabiocos** [comment]
@peruzzim FYI

**Comment by fabiocos** [comment]
please test

**Comment by perrotta** [comment]
type bugfix

**Comment by fabiocos** [comment]
please test

File: Utilities/RelMon/python/utils.py [hunk 1]
```diff
@@ -90,11 +90,20 @@ def literal2root (literal,rootType):
...
+  To be used in loops on bin number with range()
+  In case of 1-d histograms the end-of-range value should be GetNbinsX()+2 
+  In case of 2-d or 3-d histograms the product of GetNbins.()+2 gives the total number of bins, and 1 needs to be added
+  Managed using an offset variable 
+  """
+  offset=0
   biny=h.GetNbinsY()
-  if biny>1:biny+=1
+  if biny>1:
+    offset=1
```
Comment (perrotta) [inline Utilities/RelMon/python/utils.py]: PLease verify whether this offset is really needed.
In a 2D histogram there are (NBinsX+2)*(NBinsY+2) bins, including underflow and overflow. That is, they numbers from "0" to "(NBinsX+2)*(NBinsY+2)-1", and to loop through all them the second argument of the (python) range function must be "(NBinsX+2)*(NBinsY+2)", without such an offset.

Comment (fabiocos) [reply Utilities/RelMon/python/utils.py]: @perrotta right, the point is that the range loops using this output run from 0 up to the output of this function now, which must be (NBinsX+2)(NBinsY+2)+1 to account for the fact that the last element of the range() must be the last item of the list +1. Do you agree?

Comment (perrotta) [reply Utilities/RelMon/python/utils.py]: If I am not wrong,(see above)  the last item in the list is "(NBinsX+2)(NBinsY+2)-1" and the last item+1 is "(NBinsX+2)(NBinsY+2)"

Comment (fabiocos) [reply Utilities/RelMon/python/utils.py]: @perrotta no, I think you are correct, range() starts from 0 but the output of this functions counts from 1, so ok

**Comment by fabiocos** [comment]
please test

**Comment by perrotta** [comment]
+1
- Fixed as planned in the description
- Jenkins tests pass

**Comment by fabiocos** [comment]
+1

**Pull request merged**